### PR TITLE
link cut tree module

### PR DESCRIPTION
**Pull request by nwatx**

### CODE REVIEW

File: content/6_Advanced/LCT.mdx [hunk 1]
```diff
@@ -13,32 +13,315 @@ frequency: 1
...
+ 	* introducing self-loops.
+ * Time: O(\log N)
+ * Source:
+ 	* https://codeforces.com/blog/entry/53265 (Rerooting dynamic Euler tour trees)
+ 	* https://codeforces.com/blog/entry/18369 (On Euler tour trees)
+ * Verification:
+ 	* https://www.spoj.com/problems/DYNACON1/
+ */
+
+// TREAP OPERATIONS
```
Comment (SansPapyrus683) [inline content/6_Advanced/LCT.mdx:215]: maybe codesnip this?

File: content/6_Advanced/LCT.mdx [hunk 2]
```diff
@@ -13,32 +13,315 @@ frequency: 1
...
+		if(r) r->rev ^= true;
+		}
+	}
+
+	bool is_root() {
+		return p == 0 || (p->l != this && this != p->r);
+	}
+};
+
+struct lct {
```
Comment (SansPapyrus683) [inline content/6_Advanced/LCT.mdx:85]: rename this to LCT or LinkCutTree or something

File: content/6_Advanced/LCT.mdx [hunk 3]
```diff
@@ -13,32 +13,315 @@ frequency: 1
...
+## With Link-Cut Tree
+
+<!-- We can use a Link-Cut Tree to process dynamic connectivity queries in -->
+<!-- $\mathcal{O}(\log N)$ time. -->
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+struct node {
```
Comment (SansPapyrus683) [inline content/6_Advanced/LCT.mdx:58]: rename to Node

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/6_Advanced/LCT.mdx (hunk 1 above)
Comment (nwatx) [reply content/6_Advanced/LCT.mdx:215]: one time use basically, so probably not

Comment (SansPapyrus683) [reply content/6_Advanced/LCT.mdx:215]: what does that have to do w/ the codesnip

File: content/6_Advanced/LCT.mdx [hunk 4]
```diff
@@ -13,32 +13,319 @@ frequency: 1
...
+
+	Node() = default;
+
+	Node(int v) {
+		x = v;
+	}
+
+	void push() {
+		if(rev) {
+		rev = false;
```
Comment (bqi343) [inline content/6_Advanced/LCT.mdx:74]: why isn't this stuff indented

File: content/6_Advanced/LCT.mdx [hunk 5]
```diff
@@ -13,32 +13,319 @@ frequency: 1
...
 
-- [Splay Tree Paper](https://www.cs.cmu.edu/~sleator/papers/self-adjusting.pdf)
-- [Stanford Presentation](https://web.stanford.edu/class/archive/cs/cs166/cs166.1146/lectures/08/Slides08.pdf)
-
-### Implementation
+<Resources>
+	<Resource starred source="MIT" title="MIT Splay Tree Notes" url="https://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-854j-advanced-algorithms-fall-2008/lecture-notes/lec6.pdf"/>
+	<Resource
+		source="Stanford"
+		title="Stanford Splay Tree Pressentation"
```
Comment (bqi343) [inline content/6_Advanced/LCT.mdx:20]: typo

File: content/6_Advanced/LCT.mdx [hunk 6]
```diff
@@ -13,32 +13,319 @@ frequency: 1
...
-<IncompleteSection />
+## Link Cut Tree
 
-(Related?)
+<Resources>
+	<Resource starred source="MIT" title="MIT Dynamic Graphs Lecture - Link Cut Tree  and HLD" url="https://www.youtube.com/watch?v=XZLN6NxEQWo"/>
+	<Resource starred source="MIT" title="MIT Dynamic Graphs Lecture Notes" url="http://courses.csail.mit.edu/6.851/spring12/scribe/L19.pdf" />
+	<Resource source="CTSC" title="Link-Cut Tree Exploration" url="https://www.overleaf.com/read/zbbwngftkrkx?fbclid=IwAR1VJmGscfSZJHejVqU5_2Nlm8-R09pFlU7PpEg_i3CXEd0Qw-dzliSn5RI" />
+	<Resource source="CF" title="Euler Tour Tree" url="https://codeforces.com/blog/entry/18369"/>
+	<Resource
```
Comment (bqi343) [inline content/6_Advanced/LCT.mdx:38]: typo

**Review by bqi343 — CHANGES_REQUESTED** [review]

**Review by bqi343 — APPROVED** [review]

**Pull request merged**